packages/web-app/hooks/useAgreementCreate.tsx

```diff
@@ -0,0 +1,21 @@
+import { useContractWrite } from "wagmi";
+import contractInterface from "../abis/IAgreementFramework.json";
+
+const abi = contractInterface.abi;
+const contractAddress = "0xb47262C22280f361ad47Af0636086463Bd29A109";
```

Comment:
Instead of hardcoding, maybe fetch this address from a config file?

Reply:
Yeah, this code is still missing refactor, so the SDK it's outside and config is splitted from the actual pages / components